### PR TITLE
Add methods for validating aspects of PCZT bundles

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: src/pczt/verify.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            // This is not a dummy note; if the FVK field is present, it must match.
+            (Some(expected_fvk), Some(vk), _)
+                if vk.ak == expected_fvk.vk.ak && vk.nk == expected_fvk.vk.nk =>
+            {
+                Ok(vk)
+            }
+            (Some(_), Some(_), _) => Err(VerifyError::MismatchedFullViewingKey),
+            (None, Some(vk), _) => Ok(vk),
+            (None, None, _) => Err(VerifyError::MissingProofGenerationKey),
+        }
```
Comment (daira) [inline src/pczt/verify.rs:54]: The following is more obviously just logical unification with the `Some(_), Some(fvk), Some(0)` case treated as an exception (which is what the docs for `verify_nullifier` and `verify_rk` suggest to me that it should be):
```suggestion
        match (expected_fvk, vk, self.value.as_ref()) {
            (Some(expected_fvk), Some(vk), _)
                if vk.ak == expected_fvk.vk.ak && vk.nk == expected_fvk.vk.nk =>
            {
                Ok(vk)
            }
            // `expected_fvk` is ignored if the spent note is a dummy note.
            (Some(_), Some(vk), Some(value)) if value.inner() == 0 => Ok(vk),
            (Some(_), Some(_), _) => Err(VerifyError::MismatchedFullViewingKey),
            (Some(expected_fvk), None, _) => Ok(expected_fvk),
            (None, Some(vk), _) => Ok(vk),
            (None, None, _) => Err(VerifyError::MissingProofGenerationKey),
        }
```

**Review by daira — APPROVED** [review]
utACK with suggestion.

**Review by nuttycom — APPROVED** [review]
utACK 231f81911628499a8877be57e66e60c09e55bdea modulo @daira's suggestions.

**Review by daira — COMMENTED** [review]
Post-hoc ACK

**Pull request merged**